### PR TITLE
assignment 4

### PR DESCRIPTION
**Pull request by aayush-practo**
- conditional root page
- using passanger gem
- adding comment validations
- custom validations
- drying controllers
- using callback in models

### CODE REVIEW

**Comment by tarunjadhwani** [comment]
Application level polishing
- [x] don't show image when no image uploaded
- [x] show date for blog, in this format "3:15PM 12th Aug, 2016
- [x] comments bug (data shown after submit)
- [x] Add flash messages  for all rediects
- [x] Images should have a size limit

Code level polishing
- [x] root page logic in application controller
- [x] dry code for controllers
- [x] asocaition check options other then dependent destroy
- [x] custom validation
- [x] use a callback in model
- [x] Read what all files/folders does
- [x] read about prod and dev env
- [x] diff model and table name
- [x] different foreign key in table
- [x] change validation message

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -21,12 +21,11 @@ def create
   end
 
   def edit
-    @article = Article.find(params[:id])
+    find_article
```
Comment (tarunjadhwani) [inline app/controllers/articles_controller.rb]: create a after filter

File: app/models/comment.rb [hunk 2]
```diff
@@ -9,7 +9,40 @@
...
+      end
+    else
+      record.errors.add(:body, "Should be present")
+    end
+  end
+end
+
 
 class Comment < ActiveRecord::Base
+  include ActiveModel::Validations
```
Comment (tarunjadhwani) [inline app/models/comment.rb]: this is already included don't, isn't it?

File: public/assets/.sprockets-manifest-837d890e66f686a8caa844b3c5645a6d.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"files":{"application-694bcb21e3d42374420fa2c872d610c3a737c21a25464b7bb49f3706f69fc0b1.js":{"logical_path":"application.js","mtime":"2016-08-11T12:41:17+00:00","size":141384,"digest":"694bcb21e3d42374420fa2c872d610c3a737c21a25464b7bb49f3706f69fc0b1","integrity":"sha256-aUvLIePUI3RCD6LIctYQw6c3wholRkt7tJ83BvafwLE="},"application-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css":{"logical_path":"application.css","mtime":"2016-08-11T12:41:17+00:00","size":0,"digest":"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855","integrity":"sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU="}},"assets":{"application.js":"application-694bcb21e3d42374420fa2c872d610c3a737c21a25464b7bb49f3706f69fc0b1.js","application.css":"application-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855.css"}}
```
Comment (tarunjadhwani) [inline public/assets/.sprockets-manifest-837d890e66f686a8caa844b3c5645a6d.json:1]: these not to be committed on git

File: app/models/comment.rb (hunk 2 above)
Comment (aayush-practo) [inline app/models/comment.rb]: They are included, it works after removing include statement.

**Pull request merged**